core/interfaces.go

```diff
@@ -216,12 +228,15 @@ type Storage interface {
 	GetAllAppPlatforms(orgID string, appID string) ([]model.AppPlatform, error)
 
 	InsertQueueDataItemsWithContext(ctx context.Context, items []model.QueueItem) error
+	InsertQueueDataRecipientsItems(items []model.QueueItem) error
 	LoadQueueWithContext(ctx context.Context) (*model.Queue, error)
 	SaveQueueWithContext(ctx context.Context, queue model.Queue) error
 	SaveQueue(queue model.Queue) error
 
 	FindQueueData(time *time.Time, limit int) ([]model.QueueItem, error)
 	DeleteQueueData(ids []string) error
+	DeleteQueueDataForMessageWithContext(ctx context.Context, messageID string) error
+	DeleteQueueDataForMessageRecipeint(messageID string) error
```

Comment:
You have to delete for message recipient id but not for message id. Also, why do you have two functions for this?